RecoEgamma/EgammaTools/plugins/EGExtraInfoModifierFromDB.cc

```diff
@@ -82,26 +85,31 @@ class EGExtraInfoModifierFromDB : public ModifyObjectValueBase {
   std::unordered_map<unsigned,edm::Handle<edm::ValueMap<float> > > pho_vmaps;
   std::unordered_map<unsigned,edm::Handle<edm::ValueMap<int> > > pho_int_vmaps;
 
-  bool autoDetectBunchSpacing_;
-  int bunchspacing_;
-  edm::InputTag bunchspacingTag_;
-  edm::EDGetTokenT<unsigned int> bunchSpacingToken_;
+  edm::Handle<EcalRecHitCollection > ecalRecHitsEB_;
+  edm::Handle<EcalRecHitCollection > ecalRecHitsEE_;
+  edm::EDGetTokenT<EcalRecHitCollection > ecalRecHitEBToken_;
+  edm::EDGetTokenT<EcalRecHitCollection > ecalRecHitEEToken_;
+  
   float rhoValue_;
   edm::InputTag rhoTag_;
   edm::EDGetTokenT<double> rhoToken_;
-  int nVtx_;
-  edm::InputTag vtxTag_;
-  edm::EDGetTokenT<reco::VertexCollection> vtxToken_;
-  edm::Handle<reco::VertexCollection> vtxH_;
-  bool applyExtraHighEnergyProtection_;
+
+  edm::InputTag ecalRecHitEBTag_;
+  edm::InputTag ecalRecHitEETag_;
 
   const edm::EventSetup* iSetup_;
 
   std::vector<const GBRForestD*> ph_forestH_mean_;
   std::vector<const GBRForestD*> ph_forestH_sigma_; 
   std::vector<const GBRForestD*> e_forestH_mean_;
-  std::vector<const GBRForestD*> e_forestH_sigma_; 
-  const GBRForest* ep_forestH_weight_;
+  std::vector<const GBRForestD*> e_forestH_sigma_;
+
+  const CaloTopology *topology_;
+
+  bool useLocalFile_;
+  std::string addressLocalFile_;  
+  TFile* pointerLocalFile_;
```

Comment:
additional open files during execution should be avoided as much as possible.
Open the file, read what's needed from it into memory and then close the file.
Do this if memory profile allows.